pkg/hooks/webhook.go

```diff
@@ -83,7 +119,18 @@ func (w *WebhookExecutor) Call(request interface{}, response interface{}) error
 		rawRequest := json.RawMessage(reqBody)
 		logging.Logger.Info("Webhook request", "type", w.hookType, "url", w.url, "body", rawRequest)
 	}
-	resp, err := w.client.Post(w.url, "application/json", bytes.NewReader(reqBody))
+
+	var cacheEntry = w.etagCache.Get(request.GetCacheKey())
```

Comment:
important: etagCache can be nil & hence is prone to nil pointer error

Reply:
it is handled under .Get method